libs/blocks/gnav/gnav.js

```diff
@@ -384,6 +384,29 @@ class Gnav {
     return searchBar;
   };
 
+  decorateAppsMenu = async (profileEl, gnav) => {
+    const appLauncherBlock = this.body.querySelector('.app-launcher');
+    if (!appLauncherBlock) return;
+    
+    gnav.classList.add('has-apps');
+    const appLauncher = await import('./gnav-appLauncher.js');
+    const appsLink = appLauncherBlock.querySelector('a');
+    appsLink.href = makeRelative(appsLink.href, true);
+
+    const path = appsLink.href;
+    const promise = fetch(`${path}.plain.html`);
+    promise.then(async (resp) => {
+      if (resp.status === 200) {
+        const html = await resp.text();
+        const parser = new DOMParser();
+        const doc = parser.parseFromString(html, 'text/html');
+        const appList = doc.querySelectorAll('body > div > ul > li');
+
+        appLauncher.default(profileEl, appList, this.toggleMenu);
+      }
+    });
```

Comment:
I would move more of this of this into gnav-appLauncher.js. It would be nice if the HTML check and as many related bits to app launcher live in app launcher.

Reply:
Done.